app/frontend/src/features/auth/useAuthStore.test.ts

```diff
@@ -0,0 +1,368 @@
+import { renderHook } from "@testing-library/react-hooks";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { StatusCode } from "grpc-web";
+import { act } from "react-test-renderer";
+import { service, SignupArguments } from "../../service";
+import { addDefaultUser } from "../../test/utils";
+import useAuthStore, { usePersistedState } from "./useAuthStore";
+
+const getUserMock = service.user.getUser as jest.Mock;
+const getCurrentUserMock = service.user.getCurrentUser as jest.Mock;
+const updateProfileMock = service.user.updateProfile as jest.Mock;
+const updateHostingPreferenceMock = service.user
+  .updateHostingPreference as jest.Mock;
+const passwordLoginMock = service.user.passwordLogin as jest.Mock;
+const tokenLoginMock = service.user.tokenLogin as jest.Mock;
+const signupMock = service.user.completeSignup as jest.Mock;
+const getIsJailedMock = service.jail.getIsJailed as jest.Mock;
+const logoutMock = service.user.logout as jest.Mock;
+
+describe("usePersistedState hook", () => {
+  it("uses a default value", () => {
+    const defaultValue = "Test string";
+    const { result } = renderHook(() => usePersistedState("key", defaultValue));
+    expect(result.current[0]).toBe(defaultValue);
+  });
+
+  it("saves then loads a value", () => {
+    const value = "Test string";
```

Comment:
Might be worth testing this with an object value instead to check the stringifying and back works?